File: interpreter/src/main/java/cambio/tltea/interpreter/nodes/consequence/ConsequenceNode.java
package cambio.tltea.interpreter.nodes.consequence;

import cambio.tltea.interpreter.nodes.StateChangeEvent;
import cambio.tltea.interpreter.nodes.StateChangedPublisher;

import java.util.LinkedList;

/**
 * @author Lion Wagner
 */
public abstract class ConsequenceNode {

    public abstract LinkedList<EventActivationDescription> activate();

}

File: interpreter/src/main/java/cambio/tltea/interpreter/nodes/consequence/ConsequenceNode.kt
@file:JvmName("ConsequenceNode")

package cambio.tltea.interpreter.nodes.consequence

import cambio.tltea.interpreter.nodes.TriggerManager
import cambio.tltea.parser.core.temporal.TemporalOperatorInfo

/**
 * @author Lion Wagner
 */
abstract class ConsequenceNode(
    protected val triggerManager: TriggerManager,
    protected val temporalContext: TemporalOperatorInfo
) {
    abstract fun activateConsequence()
    abstract fun deactivateConsequence()
}